src/roswire/proxy/roslaunch/reader.py

```diff
@@ -23,10 +25,27 @@
 from ...name import (global_name, name_is_global, name_is_private, namespace,
                      namespace_join)
 
+
+class LaunchFileReader(abc.ABC):
+
+    @abc.abstractmethod
+    def read(self,
+             fn: str,
+             argv: Optional[Sequence[str]] = None
+             ) -> LaunchConfig:
```

Comment:
Previous docstring should be moved to here.

Reply:
You meant "copied"...see later. Done.